### DISCUSSION TITLE
GET_ALL_DEVICE_DESCRIPTIONS failed

### DISCUSSION THREAD

**Discussion by asroc-B** [SukramJ/aiohomematic#1546] [General]
### The problem

Nach dem Starten der Integration kommt der folgende Fehler:
Logger: hahomematic.client
Quelle: custom_components/homematicip_local/control_unit.py:679
Integration: Homematic(IP) Local ([Dokumentation](https://github.com/danielperna84/custom_homematic), [Probleme](https://github.com/danielperna84/hahomematic/issues))
Erstmals aufgetreten: 15:44:41 (1 Vorkommnisse)
Zuletzt protokolliert: 15:44:41
GET_ALL_DEVICE_DESCRIPTIONS failed: ExpatError [not well-formed (invalid token): line 3, column 1710]

### What version of HomematicIP (local) has the issue?

1.60.1

### What type of installation are you running?

Home Assistant OS

### What type of installation are you running for your homematic backend?

RaspberryMatic HA-Addon

### Which version of your homematic backend are you running?

3.75.7.20240420 (RaspberryMatic)

### What hardware are you running for your system?

Generic x86-64

### Which config details do you use

- [ ] TLS
- [ ] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [ ] Homematic IP
- [X] Homematic (Bidcos-RF)
- [ ] Groups (Heating-Group)
- [ ] BidCos-Wired (HM-Wired)

### Diagnostics information (no logs here)

[config_entry-homematicip_local-801237ad2d12383bc9a8f913fb3f7a02.json](https://github.com/danielperna84/hahomematic/files/15209896/config_entry-homematicip_local-801237ad2d12383bc9a8f913fb3f7a02.json)
[home-assistant_homematicip_local_2024-05-04T13-44-56.060Z.log](https://github.com/danielperna84/hahomematic/files/15209897/home-assistant_homematicip_local_2024-05-04T13-44-56.060Z.log)

Untenstehnd die Geräte im RaspberryMatic Addon (Zugriff von dort funktioniert).
![grafik](https://github.com/danielperna84/hahomematic/assets/154887920/d6fc5dc4-62ee-4438-9891-800d8b216ca2)

### Additional information

Ich habe bereits die Homematic_local Integration deinstalliert und auch das RaspberryMatic AddOn. Anschließend beides wieder komplett neu installiert und die Geräte angelernt.
Leider funktioniert der "Import" der Devices nicht nach HA

**Comment by SukramJ** [comment]
Ich gehe hier nicht von einem Fehler in der Integration aus, da der Fehler darauf hinweist, das die Antwort von der CCU einen Fehler enthält. Bisher habe ich diesen speziellen Fehler noch nicht gesehen, und kann dir somit keine Lösung anbieten.
In der Vergangenheit waren solche Fehler auf ein angelerntes Gerät zurückzuführen.

**Reply by SukramJ** [reply]
@Baxxy13 Wie kann man rausfinden, warum die XML-Antwort nicht valide ist (`ExpatError [not well-formed (invalid token): line 3, column 1710])`)

**Reply by Baxxy13** [reply]
Hmm, konkret helfen kann ich dir erstmal nicht. 
Vor allem nicht weil ich von Python keine Ahnung habe und somit auch nicht sagen kann worüber der Parser da stolpert.
Bisschen Recherche führt mich zu:

> xml.parsers.expat.errors.XML_ERROR_INVALID_TOKEN
> Raised when an input byte could not properly be assigned to a character; for example, a NUL byte (value 0) in a UTF-8 input stream. 

Ich weiß aber nicht ob das hier konkret zutrifft. 
Man bräuchte mal die komplette Rückgabe von GET_ALL_DEVICE_DESCRIPTIONS.

Wird hierbei eigentlich das  `fetch_all_device_data.fn `  oder eine XPL-RPC/JSON-RPC Methode benutzt?

**Reply by SukramJ** [reply]
> Wird hierbei eigentlich das fetch_all_device_data.fn  oder eine XPL-RPC/JSON-RPC Methode benutzt?

Nein.

Dahinter steckt nur der Aufruf von der XmlRPC-Methode `listDevices`

**Reply by Baxxy13** [reply]
Ah, ok. 
Soweit ich weiß muss `listDevices` für jedes Interface separat ausgeführt werden. 
Wie läuft das bei der Integration ab?

Mann könnte ja, zwecks Ausschlussverfahren, erstmal nur HmIP ODER BidCos aktivieren (und dann mal umgedreht) um zu gucken bei welchem Interface es hakt.

**Reply by asroc-B** [reply]
Also ich habe gestern Abend nochmal ein komplettes HA Backup eingespielt. Seither tritt das Problem nicht mehr auf. Ich kann die HM-Geräte wieder von HA aus bedienen bzw darauf zugreifen.

Vielen Dank für Eure Bemühungen und schneller Hilfe!

**Reply by Baxxy13** [reply]
Na umso besser. 😉

**Reply by SukramJ** [reply]
@Baxxy13 Vielen Dank
@asroc-B Danke für die Rückmeldung

**Reply by Baxxy13** [reply]
Was ich dann noch vorgeschlagen hätte:
Mit dem Homematic-Manager mal für jedes Interface `listDevices` ausführen. 
Der parst ja auch die XML und gibt JSON aus. 
Wäre interessant zu sehen ob der auch "gestolpert" wäre, aber gut... ist erledigt.

**Reply by asroc-B** [reply]
@Baxxy13 Sollte das Problem erneut auftreten, werde ich mich hier nochmal melden.
